### ISSUE TITLE
Not all games are shown

### ISSUE THREAD

**Issue by a-cristi** [Heroic-Games-Launcher/HeroicGamesLauncher#2526]
### Describe the bug

Using version 2.6.2 on SteamDeck I can not see all of my Epic Games games in the list of games, nor do they appear as suggestions in the search bar.

I'll try to give some context, in case it helps. First I noticed this when trying to install Borderlands 3 and searching for `Borderlands` got me no results. I have Borderlands 2, 3, and The Pre-Sequel, as well as DLC for them in my library so something should have shown up. I fixed this by logging out of my Epic account and logging back in.

I managed to install Borderlands 3 like this, but after it was installed, the game page reported that the game is not installed, and once I went back to the library screen the game was no longer there. Again, I logged out and logged back in and the game was there, but DLC was not listed when searching for it. Since then I have not managed to workaround this issue.

Periodically, I have to log out and log back in in order to see all the games. Sometimes, right after I open the launcher I can see all the games, but after a few seconds a lot of them disappear. Sometimes there is enough time until the games disappear to actually start the game, but after that the launcher still does not see the game as being in my library.

This is not a BL3 issue. After a fresh log in the launcher reports that I have 417 games, when the issue appears it reports only 229. I also have my GOG account linked, but at the moment I do not know if this issue also affects those games.

### Add logs

Logs collected after a startup in which not all games were present. The logs also include the moment I did a log out and log back in for my Epic Games account:

```text
(12:23:13) INFO:    [Connection]:       Connectivity: check-online
(12:23:13) INFO:    [Connection]:       Pinging external endpoints
(12:23:13) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(12:23:13) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(12:23:13) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(12:23:13) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(12:23:13) WARNING: [Backend]:          Failed to register protocol with OS.
(12:23:13) INFO:    [Connection]:       Connectivity: online
(12:23:13) INFO:    [Gog]:              Getting data about the user
(12:23:13) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
(12:23:13) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(12:23:13) INFO:    [Gog]:              Saved user data to config
(12:23:13) INFO:    [Backend]:          Loading Screen Ready
(12:23:13) INFO:    [Frontend]:         Refreshing Library
(12:23:13) INFO:    [Legendary]:        Refreshing library...
(12:23:13) INFO:    [Legendary]:        Refreshing Epic Games...
(12:23:14) ERROR:   [Legendary]:        TypeError: Cannot read properties of undefined (reading 'includes')
    at R.loadFile (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12724)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12998
    at Set.forEach (<anonymous>)
    at R.loadAll (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12981)
    at R.getGames (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:7656)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:184:10313
    at node:electron/js2c/browser_init:2:97569
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81082)
    at EventEmitter.emit (node:events:513:28)
(12:23:14) INFO:    [Legendary]:        Game list updated, got 148 games & DLCs
(12:23:14) INFO:    [Gog]:              Getting GOG library
(12:23:14) INFO:    [Backend]:          Frontend Ready
(12:23:14) INFO:    [Backend]:          Checking for current version changelog
(12:23:14) INFO:    [Backend]:          

Heroic Version: 2.6.2 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.6

Electron Version: 22.2.0
Chrome Version: 108.0.5359.215
NodeJS Version: 16.17.1

OS: Freedesktop SDK KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @3.06 GOVERNOR: schedutil
RAM: Total: 14.47 GiB Available: 11.75 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: x11

(12:23:14) INFO:    [Gog]:              Number of library pages: 1
(12:23:14) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:23:14) INFO:    [Gog]:              Saved games data
(12:23:15) INFO:    [Legendary]:        Checking for game updates: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:23:15) INFO:    [Winetricks]:       Downloading Winetricks
(12:23:18) INFO:    [Legendary]:        Found 0 games to update
(12:23:18) INFO:    [Gog]:              Found 0 game(s) to update
(12:23:19) INFO:    [Backend]:          Starting the Download Queue
(12:30:17) INFO:    [Frontend]:         Refreshing Library
(12:30:17) INFO:    [Legendary]:        Refreshing library...
(12:30:17) INFO:    [Legendary]:        Refreshing Epic Games...
(12:30:17) ERROR:   [Legendary]:        TypeError: Cannot read properties of undefined (reading 'includes')
    at R.loadFile (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12724)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12998
    at Set.forEach (<anonymous>)
    at R.loadAll (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12981)
    at R.getGames (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:7656)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:184:10313
    at node:electron/js2c/browser_init:2:97569
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81082)
    at EventEmitter.emit (node:events:513:28)
(12:30:17) INFO:    [Legendary]:        Game list updated, got 148 games & DLCs
(12:30:17) INFO:    [Gog]:              Getting GOG library
(12:30:18) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:30:18) INFO:    [Gog]:              Number of library pages: 1
(12:30:18) INFO:    [Gog]:              Saved games data
(12:30:19) INFO:    [Legendary]:        Checking for game updates: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:30:22) INFO:    [Legendary]:        Found 0 games to update
(12:30:22) INFO:    [Gog]:              Found 0 game(s) to update
(12:30:30) INFO:    [Legendary]:        Logging out: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --delete
(12:30:31) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary cleanup
(12:30:31) INFO:    [Frontend]:         Refreshing Library
(12:30:31) INFO:    [Legendary]:        Refreshing library...
(12:30:31) INFO:    [Gog]:              Getting GOG library
(12:30:31) INFO:    [Backend]:          Checking for current version changelog
(12:30:32) INFO:    [Gog]:              Number of library pages: 1
(12:30:34) WARNING: [Gog]:              Unable to get covers from gamesdb for Cris Tales Demo. Trying to get it from api.gog.com
(12:30:34) WARNING: [Gog]:              Couldn't get info from api.gog.com, Using fallback vertical image
(12:30:34) WARNING: [Gog]:              Unable to get covers from gamesdb for Cyberpunk 2077 Goodies Collection. Trying to get it from api.gog.com
(12:30:34) WARNING: [Gog]:              Couldn't get info from api.gog.com, Using fallback vertical image
(12:30:44) WARNING: [Gog]:              Unable to get covers from gamesdb for Warhammer Skulls Digital Goodies. Trying to get it from api.gog.com
(12:30:44) WARNING: [Gog]:              Couldn't get info from api.gog.com, Using fallback vertical image
(12:30:45) INFO:    [Gog]:              Saved games data
(12:30:45) INFO:    [Frontend]:         No cache found, getting data from legendary...
(12:30:45) INFO:    [Legendary]:        Refreshing library...
(12:30:45) INFO:    [Gog]:              Found 0 game(s) to update
(12:31:18) INFO:    [Legendary]:        Logging in: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
(12:31:20) INFO:    [Frontend]:         Refreshing Library
(12:31:20) INFO:    [Legendary]:        Refreshing library...
(12:31:20) INFO:    [Legendary]:        Refreshing Epic Games...
(12:31:20) ERROR:   [Legendary]:        Failed to parse AlewifeDLC2.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse AlewifeDLC3.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse MtMassive.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse bobcat.json
(12:31:20) INFO:    [Legendary]:        Game list updated, got 361 games & DLCs
(12:31:20) INFO:    [Frontend]:         No cache found, getting data from legendary...
(12:31:20) INFO:    [Legendary]:        Refreshing library...
(12:31:20) ERROR:   [Legendary]:        Failed to parse AlewifeDLC2.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse AlewifeDLC3.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse MtMassive.json
(12:31:20) ERROR:   [Legendary]:        Failed to parse bobcat.json
(12:31:20) INFO:    [Legendary]:        Game list updated, got 361 games & DLCs
(12:31:20) INFO:    [Frontend]:         Force Update
(12:31:21) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
```

Logs collected after I quit the launcher and started it again:

```text
(12:32:28) INFO:    [Connection]:       Connectivity: check-online
(12:32:28) INFO:    [Connection]:       Pinging external endpoints
(12:32:28) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(12:32:28) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(12:32:28) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(12:32:28) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(12:32:29) WARNING: [Backend]:          Failed to register protocol with OS.
(12:32:29) INFO:    [Connection]:       Connectivity: online
(12:32:29) INFO:    [Gog]:              Getting data about the user
(12:32:29) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
(12:32:29) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(12:32:29) INFO:    [Backend]:          Loading Screen Ready
(12:32:29) INFO:    [Frontend]:         Refreshing Library
(12:32:29) INFO:    [Legendary]:        Refreshing library...
(12:32:29) INFO:    [Legendary]:        Refreshing Epic Games...
(12:32:29) ERROR:   [Legendary]:        TypeError: Cannot read properties of undefined (reading 'includes')
    at R.loadFile (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12724)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12998
    at Set.forEach (<anonymous>)
    at R.loadAll (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:12981)
    at R.getGames (/app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:56:7656)
    at /app/bin/heroic/resources/app.asar/build/electron/main.6ace4a2c.js:184:10313
    at node:electron/js2c/browser_init:2:97569
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81082)
    at EventEmitter.emit (node:events:513:28)
(12:32:29) INFO:    [Legendary]:        Game list updated, got 148 games & DLCs
(12:32:29) INFO:    [Gog]:              Getting GOG library
(12:32:29) INFO:    [Backend]:          Frontend Ready
(12:32:29) INFO:    [Gog]:              Saved user data to config
(12:32:30) INFO:    [Backend]:          

Heroic Version: 2.6.2 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.6

Electron Version: 22.2.0
Chrome Version: 108.0.5359.215
NodeJS Version: 16.17.1

OS: Freedesktop SDK KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.67 GOVERNOR: schedutil
RAM: Total: 14.47 GiB Available: 10.74 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: x11

(12:32:30) INFO:    [Backend]:          Checking for current version changelog
(12:32:30) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:32:30) INFO:    [Gog]:              Number of library pages: 1
(12:32:31) INFO:    [Gog]:              Saved games data
(12:32:31) INFO:    [Winetricks]:       Downloading Winetricks
(12:32:32) INFO:    [Legendary]:        Checking for game updates: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(12:32:34) INFO:    [Legendary]:        Found 0 games to update
(12:32:34) INFO:    [Backend]:          Starting the Download Queue
(12:32:34) INFO:    [Gog]:              Found 0 game(s) to update
(12:32:34) INFO:    [Frontend]:         Force Update
```

### Steps to reproduce

1. Open launcher
2. Search for a game or simply scroll through the library
3. The game is not listed

### Expected behavior

All the games in my library should be listed.

### Heroic Version

Latest Stable

### System Information

- SteamOS 3.4.4.

**Comment by arielj** [comment]
@a-cristi can you try the build from this PR? https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/2533 you can try either the appimage or the flatpak

**Comment by a-cristi** [comment]
> @a-cristi can you try the build from this PR? #2533 you can try either the appimage or the flatpak

Sure, I can give it a try. Should I uninstall the version I have currently installed? Or can I just grab the appimage from [here](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/4390213025) use it while I have the stable version installed?

> I understand one of the games on your library has some missing information that we expect

Can I do anything to help with this? Maybe manually get the metadata just for that game? Or it is not relevant.

**Comment by arielj** [comment]
> Sure, I can give it a try. Should I uninstall the version I have currently installed? Or can I just grab the appimage from [here](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/4390213025) use it while I have the stable version installed?

you can use the appimage, yes

> Can I do anything to help with this? Maybe manually get the metadata just for that game? Or it is not relevant.

I'll add some log to print out the name of the game that's causing this issue so you'll have the info on the logs, would be great if you can share that game's metadata file for us to check

**Comment by arielj** [comment]
I just pushed an update to that branch, after opening heroic I imagine all games will show up and in the general logs I hope there's a line saying `No platforms info for ....` with the file that has missing information

**Comment by a-cristi** [comment]
With that build the problem is fixed: Heroic reports the same number of titles as the Epic Games Launcher does on Windows.

> I just pushed an update to that branch, after opening heroic I imagine all games will show up and in the general logs I hope there's a line saying `No platforms info for ....` with the file that has missing information

```text
(11:18:03) INFO:    [Connection]:       Connectivity: check-online
(11:18:03) INFO:    [Connection]:       Pinging external endpoints
(11:18:03) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/legendary --version
(11:18:03) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/legendary
(11:18:03) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/gogdl
(11:18:03) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/gogdl
(11:18:04) WARNING: [Backend]:          Failed to register protocol with OS.
(11:18:04) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.config/heroic/gog_store/auth.json --version
(11:18:04) INFO:    [Connection]:       Connectivity: online
(11:18:04) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(11:18:05) INFO:    [Backend]:          

Heroic Version: 2.6.3 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7

Electron Version: 22.2.0
Chrome Version: 108.0.5359.215
NodeJS Version: 16.17.1

OS: SteamOS KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.11 GOVERNOR: schedutil
RAM: Total: 14.47 GiB Available: 10.45 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: x11

(11:18:05) INFO:    [Backend]:          Loading Screen Ready
(11:18:05) INFO:    [Frontend]:         Refreshing Library
(11:18:05) INFO:    [Legendary]:        Refreshing library...
(11:18:05) INFO:    [Legendary]:        Refreshing Epic Games...
(11:18:05) WARNING: [Legendary]:        No platforms info for AlewifeDLC2.json AlewifeDLC2
(11:18:05) WARNING: [Legendary]:        No platforms info for AlewifeDLC3.json AlewifeDLC3
(11:18:05) INFO:    [Legendary]:        Game list updated, got 366 games & DLCs
(11:18:05) INFO:    [Backend]:          Frontend Ready
(11:18:05) INFO:    [Backend]:          Checking for new Heroic Updates
(11:18:05) INFO:    [Backend]:          Checking for current version changelog
(11:18:05) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:18:05) ERROR:   [Backend]:          Error when checking for current Heroic changelog
(11:18:06) INFO:    [Legendary]:        Checking for game updates: /tmp/.mount_HeroicsWobw1/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(11:18:06) INFO:    [Winetricks]:       Downloading Winetricks
(11:18:08) WARNING: [Legendary]:        Failed to parse games from /home/deck/.config/legendary/installed.json with: Error: ENOENT: no such file or directory, open '/home/deck/.config/legendary/installed.json'
    at Object.openSync (node:fs:594:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:462:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at M.listUpdateableGames (/tmp/.mount_HeroicsWobw1/resources/app.asar/build/electron/main.649562f1.js:6:11674)
    at async /tmp/.mount_HeroicsWobw1/resources/app.asar/build/electron/main.649562f1.js:185:3970
    at async node:electron/js2c/browser_init:2:97553
(11:18:08) INFO:    [Legendary]:        Found 0 games to update
(11:18:08) INFO:    [Gog]:              Found 0 game(s) to update
(11:18:10) INFO:    [Backend]:          Starting the Download Queue
```

**Comment by arielj** [comment]
thanks @a-cristi, can you share one of those files? AlewifeDLC2.json or AlewifeDLC3.json, would be useful to know what's in those files

they should be in /home/deck/.config/legendary/metadata

**Comment by a-cristi** [comment]
AlewifeDLC2.json:

```json
{
  "app_name": "AlewifeDLC2",
  "app_title": "From Dusk Till Casa Bonita - DLC",
  "asset_infos": {},
  "base_urls": [],
  "metadata": {
    "categories": [
      {
        "path": "public"
      },
      {
        "path": "hidden"
      },
      {
        "path": "games"
      },
      {
        "path": "applications"
      }
    ],
    "creationDate": "2019-11-20T20:22:16.109Z",
    "customAttributes": {
      "parentPartnerLinkId": {
        "type": "STRING",
        "value": "1636f05db30b48fd9c2d96a38cdd25d1"
      },
      "partnerLinkId": {
        "type": "STRING",
        "value": "3e226bb5046e4ce881982b823c7b9d4d"
      },
      "partnerLinkType": {
        "type": "STRING",
        "value": "ubisoft"
      }
    },
    "description": "From Dusk Till Casa Bonita - DLC",
    "developer": "Epic Dev Test Account",
    "developerId": "o-ufmrk5furrrxgsp5tdngefzt5rxdcn",
    "endOfSupport": false,
    "entitlementName": "4cb6fc07d9d6469eb00f50f9865c11ef",
    "entitlementType": "EXECUTABLE",
    "eulaIds": [
      "egstore"
    ],
    "id": "4cb6fc07d9d6469eb00f50f9865c11ef",
    "itemType": "DURABLE",
    "keyImages": [],
    "lastModifiedDate": "2019-12-20T15:33:49.356Z",
    "mainGameItem": {
      "ageGatings": {},
      "applicationId": "",
      "categories": [
        {
          "path": "public"
        },
        {
          "path": "games"
        },
        {
          "path": "applications"
        }
      ],
      "creationDate": "2019-10-28T21:43:21.118Z",
      "customAttributes": {
        "CanRunOffline": {
          "type": "STRING",
          "value": "true"
        },
        "CanSkipKoreanIdVerification": {
          "type": "STRING",
          "value": "true"
        },
        "FolderName": {
          "type": "STRING",
          "value": "SouthParkTFBWhole"
        },
        "MonitorPresence": {
          "type": "STRING",
          "value": "true"
        },
        "NeverUpdate": {
          "type": "STRING",
          "value": "true"
        },
        "PresenceId": {
          "type": "STRING",
          "value": "764eb429792649e4a606413cd649de2c"
        },
        "RequirementsJson": {
          "type": "STRING",
          "value": "[]"
        },
        "UseAccessControl": {
          "type": "STRING",
          "value": "false"
        },
        "partnerLinkId": {
          "type": "STRING",
          "value": "1636f05db30b48fd9c2d96a38cdd25d1"
        },
        "partnerLinkType": {
          "type": "STRING",
          "value": "ubisoft"
        }
      },
      "description": "South Park: Fractured But Whole",
      "developer": "Epic Dev Test Account",
      "developerId": "o-ufmrk5furrrxgsp5tdngefzt5rxdcn",
      "endOfSupport": false,
      "entitlementName": "757dcc774bbd4af1b71c450dd9e9159f",
      "entitlementType": "EXECUTABLE",
      "eulaIds": [
        "egstore"
      ],
      "id": "757dcc774bbd4af1b71c450dd9e9159f",
      "itemType": "DURABLE",
      "keyImages": [
        {
          "height": 1450,
          "md5": "4259bedb66ea53cf5b4f2ccf9e7115b3",
          "size": 877080,
          "type": "DieselGameBox",
          "uploadedDate": "2023-03-02T16:12:28.672Z",
          "url": "https://cdn1.epicgames.com/item/764eb429792649e4a606413cd649de2c/GameName_Store_Landscape_2580x1450_2580x1450-4259bedb66ea53cf5b4f2ccf9e7115b3",
          "width": 2580
        },
        {
          "height": 1600,
          "md5": "08cb8aefb48b63675e5e26eeef6d999b",
          "size": 498481,
          "type": "DieselGameBoxTall",
          "uploadedDate": "2023-03-02T16:12:31.684Z",
          "url": "https://cdn1.epicgames.com/item/764eb429792649e4a606413cd649de2c/SPFB_StorePortrait_1200x1600__1200x1600-08cb8aefb48b63675e5e26eeef6d999b",
          "width": 1200
        }
      ],
      "lastModifiedDate": "2023-03-02T16:12:39.113Z",
      "namespace": "764eb429792649e4a606413cd649de2c",
      "releaseInfo": [
        {
          "appId": "Alewife",
          "dateAdded": "2019-12-16T00:00:00.000Z",
          "id": "139bf3dc62254a6389cc85a4950114fc",
          "platform": [
            "Windows"
          ]
        }
      ],
      "selfRefundable": true,
      "status": "ACTIVE",
      "title": "South Park: Fractured But Whole",
      "unsearchable": false
    },
    "namespace": "764eb429792649e4a606413cd649de2c",
    "releaseInfo": [
      {
        "appId": "AlewifeDLC2",
        "dateAdded": "2019-11-20T00:00:00.000Z",
        "id": "f67c43f73de84b439bf5f7c3f8862856"
      }
    ],
    "status": "ACTIVE",
    "title": "From Dusk Till Casa Bonita - DLC",
    "unsearchable": false
  }
}
```

**Comment by arielj** [comment]
thanks!

next heroic release should include this fix

**Comment by arielj** [comment]
@a-cristi can you give the latest build on that PR a test? I made one change after your test and I don't own that game with the DLCs to test this

would be great if you can confirm it's working

**Comment by a-cristi** [comment]
Yes, I would be happy to help with this. Can it wait until Monday?

**Comment by arielj** [comment]
> Yes, I would be happy to help with this. Can it wait until Monday?

for sure!

**Comment by a-cristi** [comment]
Tried the latest build from that PR and everything seems fine. All games (as far as I can tell) are listed, including the problematic DLCs.

There is one error in the logs, but since I'm not using Legendary directly I guess this is fine:

```
(09:04:01) INFO:    [Connection]:       Connectivity: check-online
(09:04:01) INFO:    [Connection]:       Pinging external endpoints
(09:04:01) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/legendary --version
(09:04:02) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/legendary
(09:04:02) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/gogdl
(09:04:02) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/gogdl
(09:04:03) WARNING: [Backend]:          Failed to register protocol with OS.
(09:04:03) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.config/heroic/gog_store/auth.json --version
(09:04:03) INFO:    [Connection]:       Connectivity: online
(09:04:03) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(09:04:04) INFO:    [Backend]:          

Heroic Version: 2.6.3 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7

Electron Version: 23.1.3
Chrome Version: 110.0.5481.179
NodeJS Version: 18.12.1

OS: SteamOS KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.43 GOVERNOR: schedutil
RAM: Total: 14.47 GiB Available: 11.6 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: x11

(09:04:04) INFO:    [Backend]:          Loading Screen Ready
(09:04:04) INFO:    [Frontend]:         Refreshing Library
(09:04:04) INFO:    [Legendary]:        Refreshing library...
(09:04:04) INFO:    [Legendary]:        Refreshing Epic Games...
(09:04:04) WARNING: [Legendary]:        No platforms info for AlewifeDLC2.json AlewifeDLC2
(09:04:04) WARNING: [Legendary]:        No platforms info for AlewifeDLC3.json AlewifeDLC3
(09:04:04) INFO:    [Legendary]:        Game list updated, got 366 games & DLCs
(09:04:04) INFO:    [Backend]:          Frontend Ready
(09:04:04) INFO:    [Backend]:          Checking for new Heroic Updates
(09:04:04) INFO:    [Backend]:          Checking for current version changelog
(09:04:04) INFO:    [Winetricks]:       Downloading Winetricks
(09:04:04) ERROR:   [Backend]:          Error when checking for current Heroic changelog
(09:04:04) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(09:04:05) INFO:    [Legendary]:        Checking for game updates: /tmp/.mount_HeroicUKVIRY/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(09:04:09) INFO:    [Backend]:          Starting the Download Queue
(09:04:09) WARNING: [Legendary]:        Failed to parse games from /home/deck/.config/legendary/installed.json with: Error: ENOENT: no such file or directory, open '/home/deck/.config/legendary/installed.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at W.listUpdateableGames (/tmp/.mount_HeroicUKVIRY/resources/app.asar/build/electron/main.744f1660.js:6:11674)
    at async /tmp/.mount_HeroicUKVIRY/resources/app.asar/build/electron/main.744f1660.js:191:3970
    at async node:electron/js2c/browser_init:2:97893
(09:04:09) INFO:    [Legendary]:        Found 0 games to update
(09:04:09) INFO:    [Gog]:              Found 0 game(s) to update
```
